### PR TITLE
fix: [PIE-3732]: ui core local dev fix

### PR DESCRIPTION
**Pull request by AshwinBhatkal-Harness**
##### Summary:

Added required version hint to ui core shared dependency

##### Jira Links:

https://harness.atlassian.net/browse/PIE-3732

##### Screenshots:

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Review by ishantmehtaharness — DISMISSED** [review]
Nice catch !!

File: configs/modulefederation.config.js [hunk 1]
```diff
@@ -49,7 +49,9 @@ module.exports = ({ enableGitOpsUI, enableSTO }) => {
     name: 'nextgenui',
     remotes,
     shared: {
-      '@harness/uicore': packageJSON.dependencies['@harness/uicore'],
+      '@harness/uicore': {
```
Comment (vkbansal) [inline configs/modulefederation.config.js:52]: We should make this change only in dev mode and use the previous thing in production mode.

**Review by ishantmehtaharness — DISMISSED** [review]
LOOKS GOOD

**Review by vkbansal — DISMISSED** [review]

**Review by knagurski — APPROVED** [review]
Nice catch!

**Review by abhinavrastogi-harness — APPROVED** [review]

**Review by vkbansal — APPROVED** [review]

**Pull request merged**